src/test/scala/edu/ie3/simona/agent/participant/EvcsAgentModelCalculationSpec.scala

```diff
@@ -1987,6 +1992,425 @@ class EvcsAgentModelCalculationSpec
+          result.getQ should beEquivalentTo(0d.asMegaVar)
+      }
+
+      /* TICK 1800
+       * - ev1800 arrives
+       * - charging with 11 kW
+       */
+      scheduler.send(evcsAgent, Activation(1800))
+
+      val ev1800 = EvModelWrapper(evB.copyWithDeparture(4500))
+
+      evService.send(
+        evcsAgent,
+        ProvideEvDataMessage(
+          1800,
+          evService.ref,
+          ArrivingEvs(Seq(ev1800)),
+          Some(2700),
+        ),
+      )
+
+      scheduler.expectMsg(Completion(evcsAgent.toTyped, Some(2700)))
 
+      resultListener.expectMsgType[ParticipantResultEvent] match {
+        case ParticipantResultEvent(result: EvResult) =>
+          result.getInputModel match {
+            case model if model == ev900.uuid =>
+              result.getTime shouldBe 900.toDateTime
+              result.getP should beEquivalentTo(11d.asKiloWatt)
+              result.getQ should beEquivalentTo(0d.asMegaVar)
+              result.getSoc should beEquivalentTo(0.asPercent, 1e-2)
+          }
+      }
+
+      resultListener.expectMsgType[ParticipantResultEvent] match {
+        case ParticipantResultEvent(result: EvcsResult) =>
+          result.getInputModel shouldBe evcsInputModel.getUuid
+          result.getTime shouldBe 900.toDateTime
+          result.getP should beEquivalentTo(11d.asKiloWatt)
+          result.getQ should beEquivalentTo(0d.asMegaVar)
+      }
+
+      /* TICK 2700
+       * - ev2700 arrives
+       * - charging with 22 kW
+       */
+      scheduler.send(evcsAgent, Activation(2700))
+
+      val ev2700 = EvModelWrapper(evC.copyWithDeparture(5400))
+
+      evService.send(
+        evcsAgent,
+        ProvideEvDataMessage(
+          2700,
+          evService.ref,
+          ArrivingEvs(Seq(ev2700)),
+          None,
+        ),
+      )
+
+      scheduler.expectMsg(Completion(evcsAgent.toTyped, None))
+
+      resultListener.receiveN(2).foreach {
+        case ParticipantResultEvent(result: EvResult) =>
+          result.getInputModel match {
+            case model if model == ev900.uuid =>
+              result.getTime shouldBe 1800.toDateTime
+              result.getP should beEquivalentTo(11d.asKiloWatt)
+              result.getQ should beEquivalentTo(0d.asMegaVar)
+              result.getSoc should beEquivalentTo(4.74d.asPercent, 1e-2)
+            case model if model == ev1800.uuid =>
+              result.getTime shouldBe 1800.toDateTime
+              result.getP should beEquivalentTo(11d.asKiloWatt)
+              result.getQ should beEquivalentTo(0d.asMegaVar)
+              result.getSoc should beEquivalentTo(0.asPercent, 1e-2)
+            case model if model == ev2700.uuid =>
+              throw new RuntimeException(
+                s"Ev $model should arrive later at Evcs but it's already there."
+              )
+            case _ =>
+              throw new RuntimeException(
+                s"Unexpected case at EvcsAgentModelCalculationSpec occurred."
+              )
```

Comment:
Sure? Because when removing those and increase the number of expected results by one just for testing, the test will still be fine but in my opinion it should fail...

Reply:
Could you provide me with how you did that? I tried somehow what you said and it fails...

Reply:
Mh, I stashed what was failing yesterday, but now it works somehow. My bad.